Assets/Scripts/InventoryManager.cs

```diff
@@ -0,0 +1,135 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.InputSystem;
+
+public class InventoryManager : MonoBehaviour
+{
+    public static InventoryManager Instance;
+
+    //This will be the list that things are added and removed to via purchasing items
+    [SerializeField] private List<PlaceableObject> inventoryPlaceableObjects = new List<PlaceableObject>();
+    [SerializeField] private GameObject inventoryGUI;
+    [SerializeField] private GameObject inventoryGrid;
+    [SerializeField] private GameObject inventoryItemPrefab;
+    [SerializeField] private GameObject playerHeldItemParent;
+    [SerializeField] private float stockScale;
+    [SerializeField] private float structureScale;
+
+
+    private PlaceableObject heldObject;
+    public PlaceableObject HeldObject { get => heldObject; set => heldObject = value; }
+    private int tabIndex;
+    private List<GameObject> gridObjectDisplayList = new List<GameObject>();
+    //List of items currently displayed in GUI
+    private List<PlaceableObject> inventoryObjectDisplayList = new List<PlaceableObject>();
+    private GameObject playerHeldItem;
+    public GameObject PlayerHeldItem { get => playerHeldItem; set => playerHeldItem = value; }
+
+    // Makes InventorManager a singleton
+    private void Awake()
+    {
+        if (Instance != null)
+        {
+            Destroy(this);
+            return;
```

Comment:
would this not destroy the instance, then continue running the program, without making the instance again?